L1Trigger/L1TMuonEndCap/src/PtAssignmentEngineDxy.cc

```diff
@@ -0,0 +1,262 @@
+  dPhi_24 = (data.delta_ph[4] != invalid_dphi) ? data.delta_ph[4] * (data.sign_ph[4] ? 1 : -1) : 0;
+  dPhi_34 = (data.delta_ph[5] != invalid_dphi) ? data.delta_ph[5] * (data.sign_ph[5] ? 1 : -1) : 0;
+
+  // Calculate delta theta
+  dTh_12 = (data.delta_th[0] != invalid_dtheta) ? data.delta_th[0] * (data.sign_th[0] ? 1 : -1) : 0;
+  dTh_13 = (data.delta_th[1] != invalid_dtheta) ? data.delta_th[1] * (data.sign_th[1] ? 1 : -1) : 0;
+  dTh_14 = (data.delta_th[2] != invalid_dtheta) ? data.delta_th[2] * (data.sign_th[2] ? 1 : -1) : 0;
+  dTh_23 = (data.delta_th[3] != invalid_dtheta) ? data.delta_th[3] * (data.sign_th[3] ? 1 : -1) : 0;
+  dTh_24 = (data.delta_th[4] != invalid_dtheta) ? data.delta_th[4] * (data.sign_th[4] ? 1 : -1) : 0;
+  dTh_34 = (data.delta_th[5] != invalid_dtheta) ? data.delta_th[5] * (data.sign_th[5] ? 1 : -1) : 0;
+
+  // Set dPhi and dTheta values to 0 if there was no hit in the station
+  if (!st1) {
+    dPhi_12 = 0;
+    dPhi_13 = 0;
+    dPhi_14 = 0;
+
+    dTh_12 = 0;
+    dTh_13 = 0;
+    dTh_14 = 0;
+  }
+  if (!st2) {
+    dPhi_12 = 0;
+    dPhi_23 = 0;
+    dPhi_24 = 0;
+
+    dTh_12 = 0;
+    dTh_23 = 0;
+    dTh_24 = 0;
+  }
+  if (!st3) {
+    dPhi_13 = 0;
+    dPhi_23 = 0;
+    dPhi_34 = 0;
+
+    dTh_13 = 0;
+    dTh_23 = 0;
+    dTh_34 = 0;
+  }
+  if (!st4) {
+    dPhi_14 = 0;
+    dPhi_24 = 0;
+    dPhi_34 = 0;
+
+    dTh_14 = 0;
+    dTh_24 = 0;
+    dTh_34 = 0;
+  }
+
+  // Set NN inputs
+
+  // NN was trained with the wrong sign convention. TO BE CHANGED LATER!
+  x_dphi[0] = dPhi_12;
+  x_dphi[1] = dPhi_13;
+  x_dphi[2] = dPhi_14;
+  x_dphi[3] = dPhi_23;
+  x_dphi[4] = dPhi_24;
+  x_dphi[5] = dPhi_34;
+
+  // NN was trained with the wrong sign convention. TO BE CHANGED LATER!
+  x_dtheta[0] = dTh_12;
+  x_dtheta[1] = dTh_13;
+  x_dtheta[2] = dTh_14;
+  x_dtheta[3] = dTh_23;
+  x_dtheta[4] = dTh_24;
+  x_dtheta[5] = dTh_34;
+
+  // NN was trained with the wrong sign convention. TO BE CHANGED LATER!
+  x_bend_emtf[0] = bend_1;
+  x_bend_emtf[1] = bend_2;
+  x_bend_emtf[2] = bend_3;
+  x_bend_emtf[3] = bend_4;
+
+  x_fr_emtf[0] = fr_1;
+  x_trk_theta[0] = track.Theta_fp();
+  x_me11ring[0] = St1_ring2;
+
+  x_rpcbit[0] = rpc_1;
+  x_rpcbit[1] = rpc_2;
+  x_rpcbit[2] = rpc_3;
+  x_rpcbit[3] = rpc_4;
+
+  feature = {{x_dphi[0],      x_dphi[1],      x_dphi[2],      x_dphi[3],      x_dphi[4],    x_dphi[5],
+              x_dtheta[0],    x_dtheta[1],    x_dtheta[2],    x_dtheta[3],    x_dtheta[4],  x_dtheta[5],
+              x_bend_emtf[0], x_bend_emtf[1], x_bend_emtf[2], x_bend_emtf[3], x_fr_emtf[0], x_trk_theta[0],
+              x_me11ring[0],  x_rpcbit[0],    x_rpcbit[1],    x_rpcbit[2],    x_rpcbit[3]}};
+  return;
+}
+
+void PtAssignmentEngineDxy::call_tensorflow_dxy(const emtf::Feature& feature, emtf::Prediction& prediction) const {
+  static tensorflow::Tensor input(tensorflow::DT_FLOAT, {1, emtf::NUM_FEATURES});
+  static std::vector<tensorflow::Tensor> outputs;
```

Comment:
These are not thread safe.

Reply:
I just removed those 'static' lines and I'm now testing again. Thanks for the input!